### PR TITLE
Ft signin #164718857

### PR DESCRIPTION
**Pull request by HabibSentongo**
This PR merges the sign-in user branch to develop
Heroku link: https://epicmail-sentongo.herokuapp.com/
@bekeplar @kyakusahmed @jenny @neelxie @nanfuka @Wabuluka

### CODE REVIEW

**Comment by bekeplar** [comment]
My brother good work. Though you need to improve on your coverage

**Comment by HabibSentongo** [comment]
Thanks please, it's what I am working on right now

**Comment by nanfuka** [comment]
great work Sentongo

**Comment by HabibSentongo** [comment]
Thanks @nanfuka

File: app/controllers/controller.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+                                '02 : POST /api/v1/auth/login',
+                                '03 : POST /api/v1/messages',
+                                '04 : GET /api/v1/messages',
+                                '05 : GET /api/v1/messages/unread',
+                                '06 : GET /api/v1/messages/sent',
+                                '07 : GET /api/v1/messages/<message-id>',
+                                '08 : DELETE /api/v1/messages/<message-id>']
+                                    })
+    def traverse(self, uID, cID, criteria, key):
+        current_Uid = request.get_json()
```
Comment (byarustev) [inline app/controllers/controller.py:21]: I would prefer you choosing between snake style and camel case not using both

File: app/controllers/controller.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+                    return jsonify({
+                        'status': 200,
+                        'data': [Static_strings.msg_deleted]
+                    })
+        return jsonify({
+            'status': 204,
+            'error': Static_strings.error_missing
+        })
+
+    def mail_send(self):
```
Comment (byarustev) [inline app/controllers/controller.py:95]: change this to **send_email** methods should be named in form of an instruction. say a method returns a name can be named as **get_name**

File: app/controllers/controller.py [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+                'status': 302,
+                'data': selected
+                })
+        else:
+            return jsonify({
+                'status': 204,
+                'error': Static_strings.error_empty
+            })
+
+    def mail_deletor(self, key):
```
Comment (byarustev) [inline app/controllers/controller.py:76]: change this to **delete_email** methods should be named in the form of instruction. say a method returns a name can be named as get_name

File: app/controllers/controller.py [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+            })
+        if mail_details['sen_status']=='sent' and 'recieverId' not in mail_details:
+            return jsonify({
+                'status': 417,
+                'error': Static_strings.error_missdestination
+            })
+        
+        subject = mail_details.get("subject")
+        parentMessageId = mail_details.get("parentMessageId")
+        sen_status = mail_details.get("sen_status")
```
Comment (byarustev) [inline app/controllers/controller.py:115]: I see some variables are using camel case and this one is using snake style. go through your whole code base and change this to a uniform one.

File: app/controllers/controller.py [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+            new_mail.mail_struct()
+        )
+        return jsonify({
+            'status': 201,
+            'data': [
+                new_mail.mail_struct()
+            ]
+        })
+
+    def accreate(self):
```
Comment (byarustev) [inline app/controllers/controller.py:139]: what does this word mean ?? accreate change it to a name that communicates what the method does.

File: app/controllers/controller.py [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+            email_add=email_add,
+            first_name= first_name,
+            last_name= last_name,
+            password= password
+        )
+
+        user_list.append(
+            new_user.user_struct()
+        )
+        tok = create_access_token(new_user.user_struct()['user_id'])
```
Comment (byarustev) [inline app/controllers/controller.py:167]: rename this to token. token is not so long to be cut short. 
you should ensure that your code is written in a way that it makes it easy for a person to understand what is taking place. though your variable naming and method naming

**Review by byarustev — CHANGES_REQUESTED** [review]

File: app/controllers/controller.py (hunk 6 above)
Comment (HabibSentongo) [reply app/controllers/controller.py:167]: Alright I'm on it

File: app/controllers/controller.py (hunk 5 above)
Comment (HabibSentongo) [reply app/controllers/controller.py:139]: I ```meant account_create```. let me change it

File: app/controllers/controller.py (hunk 4 above)
Comment (HabibSentongo) [reply app/controllers/controller.py:115]: I am going to stick to snake style

File: app/controllers/controller.py (hunk 2 above)
Comment (HabibSentongo) [reply app/controllers/controller.py:95]: Alright

File: app/controllers/controller.py (hunk 3 above)
Comment (HabibSentongo) [reply app/controllers/controller.py:76]: Alright

File: app/controllers/controller.py (hunk 1 above)
Comment (HabibSentongo) [reply app/controllers/controller.py:21]: I am going to stick to snake style

Comment (HabibSentongo) [reply app/controllers/controller.py:21]: This has been resolved

File: app/controllers/controller.py (hunk 3 above)
Comment (HabibSentongo) [reply app/controllers/controller.py:76]: This has been resolved

File: app/controllers/controller.py (hunk 2 above)
Comment (HabibSentongo) [reply app/controllers/controller.py:95]: This has been resolved

File: app/controllers/controller.py (hunk 4 above)
Comment (HabibSentongo) [reply app/controllers/controller.py:115]: This has been resolved

File: app/controllers/controller.py (hunk 5 above)
Comment (HabibSentongo) [reply app/controllers/controller.py:139]: This has been resolved

File: app/controllers/controller.py (hunk 6 above)
Comment (HabibSentongo) [reply app/controllers/controller.py:167]: This has been resolved

**Comment by HabibSentongo** [comment]
Hello there,

I just signed the petition "Africa: Support Dr Tedros, WHO Director
General. No Resignation! Keep up the work." and wanted to see if you could
help by adding your name.

Our goal is to reach 2,500 signatures and we need more support. You can
read more and sign the petition here:

http://chng.it/jFQZpmrfnr

Thanks!
Sentongo

**Comment by Wabuluka** [comment]
Man so u used this PR

**Comment by nanfuka** [comment]
bulijjo oli ku epiki..... vwaawo naawe. haaji omulamba wano tovangawo


*KALUNGI DEBORAH NSUBUGA*
* CIU- REFACTORY <https://www.refactory.ug/>*
*Muyenga Tankhill*
*tel: +256-703-384258*
*Full stack developer*
*MY WORK <https://github.com/nanfuka>*

**Comment by HabibSentongo** [comment]
Hello there,

I just signed the petition "H​.​E. Yoweri Kaguta Museveni.,Hon. Kadaga
Rebecca, Uganda, Hon. Ruhakana Rugunda: Allocation of 20M to each Mp should
instead go support health centers, doctors and nurses." and wanted to see
if you could help by adding your name.

Our goal is to reach 15,000 signatures and we need more support. You can
read more and sign the petition here:

http://chng.it/hk5CSYBNTN

Thanks!
Sentongo

**Pull request merged**